src/library/setlogfeature.cpp

```diff
@@ -332,21 +332,24 @@ void SetlogFeature::slotPlaylistTableChanged(int playlistId) {
     }
 }
 
-void SetlogFeature::slotPlaylistContentChanged(int playlistId) {
+void SetlogFeature::slotPlaylistContentChanged(QSet<int> playlistIds) {
```

Comment:
const QSet<int>& ? 
Why is this is a QSet by the way? The hash code look up is not used.

Reply:
It is used by invoking QSet::contains(). A QList would otherwise contain duplicate track ids if we would not deduplicate it before emitting the signal. Please refer to the code that emits this signal. A QSet is a pragmatic, good enough solution that doesn't require any extra code. It is also used for similar signals in Mixxx. I refuse to optimize this away.

Reply:
OK, I did not get this before.

Reply:
The decision passing an implicit shared class by const-ref or by-value is a never ending story. The implicit sharing was designed to allow passing everything by value and keep the code simple and readable even for C++ novices.

For signals I usually prefer passing by value, because the parameters must be passed by value when connections are queued. This ensures that there is only a single code path, even if it is not the most efficient solution in all use cases.

I don't really care, but I neither want to think nor argue about each and every usage pattern. Both are correct and valid.

Reply:
If one pattern is preferred than there should be a tool to find and fix all unintended usages. Otherwise it's just mental overhead.